src/main/java/net/dries007/tfc/common/blocks/devices/ChannelBlock.java

```diff
@@ -0,0 +1,230 @@
+                    isAdjacentConnectable = true;
+                    break;
+                }
+                else if (!blockState.isAir())
+                {
+                    break;
+                }
+            }
+
+            state = state.setValue(DirectionPropertyBlock.getProperty(dir), isAdjacentConnectable);
+        }
+
+        return state;
+    }
+
+    @Override
+    public boolean onDestroyedByPlayer(BlockState state, Level level, BlockPos pos, Player player, boolean willHarvest,
+            FluidState fluid)
+        {
+        // On destroy, notify source channel that all flows going through this
+        // channel have been broken
+        if (!level.isClientSide())
+        {
+            level.getBlockEntity(pos, TFCBlockEntities.CHANNEL.get()).ifPresent(
+                    channel -> channel.notifyBrokenLink(channel.getNumberOfFlows()));
+        }
+
+        return super.onDestroyedByPlayer(state, level, pos, player, willHarvest, fluid);
+    }
+
+    @Override
+    public void onBlockExploded(BlockState state, Level level, BlockPos pos, Explosion explosion)
+    {
+        // On destroy, notify source channel that all flows going through this
+        // channel have been broken
+        if (!level.isClientSide())
+        {
+            level.getBlockEntity(pos, TFCBlockEntities.CHANNEL.get()).ifPresent(
+                    channel -> channel.notifyBrokenLink(channel.getNumberOfFlows()));
+        }
+
+        super.onBlockExploded(state, level, pos, explosion);
+    }
+
+    public ChannelBlock(ExtendedProperties properties)
+    {
+        super(properties);
+        this.registerDefaultState(
+                this.defaultBlockState()
+                        .setValue(EAST, false)
+                        .setValue(SOUTH, false)
+                        .setValue(WEST, false)
+                        .setValue(NORTH, false)
+                        .setValue(DOWN, false)
+                        .setValue(TRIGGERED, false)
+                        .setValue(WITH_METAL, false));
+    }
+
+    @Override
+    public VoxelShape getShape(BlockState state, BlockGetter level, BlockPos pos, CollisionContext context)
+    {
+        return SHAPES[(state.getValue(NORTH) ? 1 << Direction.NORTH.get2DDataValue() : 0) |
+                (state.getValue(EAST) ? 1 << Direction.EAST.get2DDataValue() : 0) |
+                (state.getValue(SOUTH) ? 1 << Direction.SOUTH.get2DDataValue() : 0) |
+                (state.getValue(WEST) ? 1 << Direction.WEST.get2DDataValue() : 0)];
+    }
+
+    @Override
+    public BlockState updateShape(BlockState state, Direction direction, BlockState adjacentState, LevelAccessor level,
+            BlockPos pos, BlockPos adjacentPos)
+    {
+        return updateConnectedSides(level, pos, state);
+    }
+
+    @Override
+    public void neighborChanged(BlockState state, Level level, BlockPos pos, Block neighborBlock, BlockPos neighborPos,
+            boolean movedByPiston)
+    {
```

Comment:
We should discuss on discord but I don't love that the channels appear to be redstone triggered when nothing in TFC has that functionality. I'm not sold on it, but I'm not sure what else we'd do for that.